src/main/java/net/adoptium/DownloadResource.java

```diff
@@ -0,0 +1,44 @@
+package net.adoptium;
+
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import io.quarkus.qute.CheckedTemplate;
+import io.quarkus.qute.TemplateInstance;
+import net.adoptium.api.DownloadRepository;
+import net.adoptium.model.DownloadResourceHTMLData;
+import net.adoptium.utils.DownloadArgumentGroup;
+import net.adoptium.utils.DownloadStringArgumentExtractor;
+import org.jboss.logging.Logger;
+import org.jboss.resteasy.annotations.jaxrs.PathParam;
+
+import java.util.Map;
+
+import net.adoptopenjdk.api.v3.models.Binary;
+
+@Path("/download")
+public class DownloadResource {
+    private static final Logger LOG = Logger.getLogger(DownloadResource.class);
+
+    @CheckedTemplate
+    public static class Templates {
+        public static native TemplateInstance download(DownloadResourceHTMLData htmlData);
```

Comment:
Can you please add some doc to this class & native method

Reply:
added javadoc